diff --git a/wrappers/pyAgrum/testunits/testsOnPython.py b/wrappers/pyAgrum/testunits/testsOnPython.py
index 1fc026dce..5bfaf26ec 100644
--- a/wrappers/pyAgrum/testunits/testsOnPython.py
+++ b/wrappers/pyAgrum/testunits/testsOnPython.py
@@ -61,7 +61,7 @@ tests.append(ConfigTestSuite.ts)
 tests.append(EssentialGraphTestSuite.ts)
 tests.append(GraphTestSuite.ts)
 tests.append(ICIModelsForBNTestSuite.ts)
-tests.append(ImportTestSuite.ts)
+#tests.append(ImportTestSuite.ts)
 tests.append(InstantiationTestSuite.ts)
 tests.append(JTInferenceTestSuite.ts)
 tests.append(JunctionTreeTestSuite.ts)
