diff --git a/wrappers/pyAgrum/testunits/gumTest.py b/wrappers/pyAgrum/testunits/gumTest.py
index aa0131edf..48af92482 100644
--- a/wrappers/pyAgrum/testunits/gumTest.py
+++ b/wrappers/pyAgrum/testunits/gumTest.py
@@ -12,7 +12,7 @@ FORMAT = '[pyAgrum] %(asctime)s | %(levelname)s | %(filename)s:%(lineno)d | %(fu
 
 log = logging.getLogger("gumTestLog")
 log.setLevel(logging.DEBUG)  # better to have too much log than not enough
-fh = logging.FileHandler('../../pyAgrumTests.log', mode='w', encoding=None, delay=False)
+fh = logging.FileHandler('/tmp/pyAgrumTests.log', mode='w', encoding=None, delay=False)
 fh.setFormatter(logging.Formatter(FORMAT))
 log.addHandler(fh)
 log.propagate = False
@@ -86,7 +86,7 @@ os.chdir(cwd)
 print("-" * 70)
 print(" log file ")
 print("-" * 70)
-with open("../../pyAgrumTests.log", "r") as logfile:
+with open("/tmp/pyAgrumTests.log", "r") as logfile:
   for f in logfile.readlines():
     if "[pyAgrum]" in f:
       print(f, end='')
